File: beecrowd/javascript/beginner/1828-bazinga.js
const input = require('fs').readFileSync('/dev/stdin', 'utf8');
const lines = input.split('\n');

const game = [
  ['tesoura', 'papel'],
  ['papel', 'pedra'],
  ['pedra', 'lagarto'],
  ['lagarto', 'Spock'],
  ['Spock', 'tesoura'],
  ['tesoura', 'lagarto'],
  ['lagarto', 'papel'],
  ['papel', 'Spock'],
  ['Spock', 'pedra'],
  ['pedra', 'tesoura'],
];

const n = Number(lines.shift());

for (var i = 1; i <= n; i++) {
  const inputs = lines.shift().split(" ");

  const a = inputs[0];
  const b = inputs[1];

  if (a === b) {
    console.log(`Caso #${i}: De novo!`);
  } else {
    for (const possibility of game) {
      if (possibility[0] === a && possibility[1] === b) {
        console.log(`Caso #${i}: Bazinga!`);
      } else if (possibility[1] === a && possibility[0] === b) {
        console.log(`Caso #${i}: Raj trapaceou!`);
      }
    }
  }
}

File: beecrowd/dart-2.9/beginner/1828-bazinga.dart
import 'dart:convert';
import 'dart:io';

const game = <List<String>>[
  ['tesoura', 'papel'],
  ['papel', 'pedra'],
  ['pedra', 'lagarto'],
  ['lagarto', 'Spock'],
  ['Spock', 'tesoura'],
  ['tesoura', 'lagarto'],
  ['lagarto', 'papel'],
  ['papel', 'Spock'],
  ['Spock', 'pedra'],
  ['pedra', 'tesoura'],
];

void main() {
  final n = _getIntLine();

  for (var i = 1; i <= n; i++) {
    final inputs = _getStringListLine();

    final a = inputs[0];
    final b = inputs[1];

    if (a == b) {
      print('Caso #$i: De novo!');
    } else {
      for (final possibility in game) {
        if (possibility[0] == a && possibility[1] == b) {
          print('Caso #$i: Bazinga!');
        } else if (possibility[1] == a && possibility[0] == b) {
          print('Caso #$i: Raj trapaceou!');
        }
      }
    }
  }
}

String _getLine() =>
    stdin.readLineSync(encoding: Encoding.getByName('utf-8'))?.trim() ?? '';

List<String> _getStringListLine() => _getLine().split(' ');

int _getIntLine() => int.parse(_getLine());
